### ISSUE TITLE
[DNN/ONNX] Squeeze operator fails when axes is empty

### ISSUE THREAD

**Issue by Anemptyship** [opencv/opencv#28424]
### System Information

- **OpenCV Version**: 4.13.0
- **OS**: Linux
- **Python**: 3.10
- **ONNX Opset**: 11
- **Comparison**: ONNX Runtime (works correctly)

### Detailed description

OpenCV's DNN module incorrectly handles the ONNX `Squeeze` operator when all axes are squeezed (i.e., removing all dimensions of size 1). While ONNX Runtime correctly produces a `[64, 64]` output tensor, OpenCV DNN retains the original `[1, 1, 64, 64]` shape.

### Expected Behavior
When a `Squeeze` node removes all size-1 dimensions:
- **Input shape**: `[1, 1, 64, 64]`
- **Expected output**: `[64, 64]`

### Actual Behavior (OpenCV DNN)
- **Actual output**: `[1, 1, 64, 64]` (no dimensions removed)

### ONNX Runtime Behavior (Reference)
- **Output**: `[64, 64]` ✓ (correct)

## Steps to Reproduce

```python
import torch
import torch.nn as nn
import cv2
import onnxruntime as ort

class SqueezeAllDims(nn.Module):
    def forward(self, x):
        return torch.squeeze(x)  # Remove all size-1 dims

model = SqueezeAllDims()
model.eval()

input_tensor = torch.randn(1, 1, 64, 64)

# Export to ONNX
torch.onnx.export(model, input_tensor, "squeeze_bug.onnx",
                 input_names=['input'], output_names=['output'],
                 opset_version=11)

# ONNX Runtime (correct)
ort_session = ort.InferenceSession("squeeze_bug.onnx")
ort_out = ort_session.run(None, {'input': input_tensor.numpy()})[0]
print(f"ONNX Runtime shape: {ort_out.shape}")  # (64, 64) ✓

# OpenCV DNN (bug)
net = cv2.dnn.readNetFromONNX("squeeze_bug.onnx")
net.setInput(input_tensor.numpy())
cv_out = net.forward()
print(f"OpenCV DNN shape: {cv_out.shape}")  # (1, 1, 64, 64) ✗
```

## Analysis

The issue likely stems from the `Squeeze` layer implementation in OpenCV DNN:
- **File**: `modules/dnn/src/layers/reshape_layer.cpp` (SqueezeLayer)
- **Root cause**: When the `axes` attribute is empty or not specified, the layer should squeeze **all** dimensions of size 1, but OpenCV appears to skip this operation

### ONNX Specification
According to [ONNX Squeeze operator spec](https://onnx.ai/onnx/operators/onnx__Squeeze.html):
> If `axes` is not provided, all the single dimensions will be removed from the shape.

OpenCV DNN does not correctly implement this fallback behavior.

## Proposed Fix

In `modules/dnn/src/layers/reshape_layer.cpp`, the `Squeeze` layer should:
1. Check if `axes` parameter is empty
2. If empty, automatically detect and remove all dimensions where `size == 1`
3. Update output shape accordingly

Example pseudocode:
```cpp
if (axes.empty()) {
    // Squeeze ALL dimensions of size 1
    for (int i = 0; i < input_shape.size(); i++) {
        if (input_shape[i] != 1) {
            output_shape.push_back(input_shape[i]);
        }
    }
} else {
    // Squeeze only specified axes
    // ... existing logic ...
}
```
## Attachments
- ONNX file: `squeeze_bug.onnx.zip`
[squeeze_bug.onnx.zip](https://github.com/user-attachments/files/24695873/squeeze_bug.onnx.zip)

## Impact
This affects any model using `torch.squeeze()` without axis specification, including:
- Models with dynamic batch sizes normalized to 1
- Auto-encoder bottleneck layers
- Channel-wise global pooling outputs

**Note**: I'm ready to submit a PR if this issue is confirmed. Please advise on the best approach.

### Issue submission checklist

- [x] I report the issue, it's not a question
- [x] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [x] I updated to the latest OpenCV version and the issue is still there
- [x] There is reproducer code and related data files (videos, images, onnx, etc)
